### DISCUSSION TITLE
Default depthCompareOp in DepthStencilState

### DISCUSSION THREAD

**Discussion by franciszhi** [vsg-dev/VulkanSceneGraph#851] [General]
Since https://github.com/vsg-dev/VulkanSceneGraph/pull/393, reverse depth convention is adopted. For that, `depthCompareOp` is changed from `VK_COMPARE_OP_LESS` to `VK_COMPARE_OP_GREATER`.

In my opinion, `VK_COMPARE_OP_GREATER_OR_EQUAL` should be used by default as it is the exact reverse of `VK_COMPARE_OP_LESS`. By doing this, when two objects of same size, same depth are drawn, the visible one should be the the last object commanded to be drawn.

**Comment by robertosfield** [comment]
The equivalent of  VK_COMPARE_OP_LESS with standard increasing depth is VK_COMPARE_OP_GREATER with reverse depth.

One can debate whether _OR_EQUAL is desirable for either case, what is most appropriate all depends upon the specifics of the model you are working with, some applications may find it better, some might find it worse for their models.

What we have is just defaults, you can create objects and if the defaults aren't appropriate you modify them to fit the purpose.
